boards/shared/Src/App/App_SharedSignal.c

```diff
@@ -0,0 +1,75 @@
+#include <assert.h>
+#include <stdlib.h>
+#include "App_SharedSignal.h"
+
+struct Signal
+{
+    // The last time this signal was low, in milliseconds
+    uint32_t last_time_low_ms;
+
+    // The last time this signal was high, in milliseconds
+    uint32_t last_time_high_ms;
+
+    // The function to call to check if this signal is high
+    bool (*is_high)(void);
+
+    // How long this signal must be continuously high for, in milliseconds,
+    // before the callback function is triggered
+    uint32_t duration_high_ms;
```

Comment:
``` suggestion
    uint32_t duration_high_ms_before_callback_triggered
```
And similarly for function args.

Reply:
That seemed a bit verbose but I see how the current name is unclear. I will fall back to using the `struct SignalCallback` that you suggested in the beginning.

i.e.,
```
struct SignalCallback {
  uint32_t high_duration_ms;
  void (*function)(void)
};
```